include/seqan3/alignment/pairwise/edit_distance_unbanded.hpp

```diff
@@ -1040,41 +1040,51 @@ class edit_distance_unbanded :
     template <typename callback_t>
     void operator()(size_t const idx, callback_t && callback)
     {
+        using traits_type = alignment_configuration_traits<align_config_t>;
         using result_value_type = typename alignment_result_value_type_accessor<alignment_result_type>::type;
 
         compute();
-        result_value_type res_vt{};
-        res_vt.id = idx;
-        if constexpr (compute_score)
-        {
-            res_vt.score = this->score().value_or(matrix_inf<score_type>);
-        }
+
+        // First cache the begin and end positions if enabled by the edit distance traits.
+        // Note, that they might be activated even if the user did not configured them, but in order to
+        // compute for example the alignment both information are required and enabled internally.
+        auto cached_end_positions = this->invalid_coordinate();
+        auto cached_begin_positions = this->invalid_coordinate();
 
         if constexpr (compute_end_positions)
-        {
-            res_vt.end_positions = this->end_positions();
-        }
+            cached_end_positions = this->end_positions();
 
         if constexpr (compute_begin_positions)
-        {
             if (this->is_valid())
-                res_vt.begin_positions = alignment_begin_positions(this->trace_matrix(), res_vt.end_positions);
-            else
-                res_vt.begin_positions = this->invalid_coordinate();
-        }
+                cached_begin_positions = alignment_begin_positions(this->trace_matrix(), cached_end_positions);
```

Comment:
What if `if constexpr (compute_end_positions)` before evaluated to false and the end positions have not been properly initialised?

Reply:
Then, someone configured the edit distance wrong. This is an internal dependency. I will add a static_assert to make sure that this is safe at compile time.